src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/x4decoder/HaivisionX4DecoderCommunicator.java

```diff
@@ -1314,94 +1435,1245 @@ public boolean mapSwitchControlValue(String value) {
+				SRTMode srtMode = SRTMode.getByName(value);
+				removeUnusedStatsAndControlBySRTMode(stats, advancedControllableProperties, createStream, streamControllingGroup);
+				createStream.setSrtMode(srtMode.getCode());
+				populateCreateStreamControl(stats, advancedControllableProperties, createStream, streamControllingGroup);
+				populateLocalExtendedStats(stats, advancedControllableProperties);
+				break;
+			case LATENCY:
+				createStream.setLatency(Integer.parseInt(value));
+				addAdvanceControlProperties(advancedControllableProperties,
+						createNumeric(stats, streamControllingGroup + StreamControllingMetric.LATENCY.getName(), checkForNullData(createStream.getLatency().toString())));
+				populateCancelButtonForCreateStream(stats, advancedControllableProperties);
+				populateLocalExtendedStats(stats, advancedControllableProperties);
+				break;
+			case SRT_TO_UDP_STREAM_CONVERSION:
+				Boolean srtToUDP = mapSwitchControlValue(value);
+				removeUnusedStatsAndControlByStreamConversion(stats, advancedControllableProperties, createStream, streamControllingGroup);
+				createStream.setSrtToUdp(srtToUDP);
+				populateCreateStreamControl(stats, advancedControllableProperties, createStream, streamControllingGroup);
+				populateLocalExtendedStats(stats, advancedControllableProperties);
+				break;
+			case SRT_TO_UDP_ADDRESS:
+				createStream.setSrtToUdpAddress(value);
+				addAdvanceControlProperties(advancedControllableProperties,
+						createText(stats, streamControllingGroup + StreamControllingMetric.SRT_TO_UDP_ADDRESS.getName(), createStream.getSrtToUdpAddress()));
+				populateCancelButtonForCreateStream(stats, advancedControllableProperties);
+				populateLocalExtendedStats(stats, advancedControllableProperties);
+				break;
+			case SRT_TO_UDP_PORT:
+				createStream.setSrtToUdpPort(Integer.parseInt(value));
+				addAdvanceControlProperties(advancedControllableProperties,
+						createNumeric(stats, streamControllingGroup + StreamControllingMetric.SRT_TO_UDP_PORT.getName(), checkForNullData(createStream.getSrtToUdpPort().toString())));
+				populateCancelButtonForCreateStream(stats, advancedControllableProperties);
+				populateLocalExtendedStats(stats, advancedControllableProperties);
+				break;
+			case SRT_TO_UDP_TOS:
+				createStream.setSrtToUdpTos(value);
+				addAdvanceControlProperties(advancedControllableProperties,
+						createText(stats, streamControllingGroup + StreamControllingMetric.SRT_TO_UDP_TOS.getName(), createStream.getSrtToUdpTos()));
+				populateCancelButtonForCreateStream(stats, advancedControllableProperties);
+				populateLocalExtendedStats(stats, advancedControllableProperties);
+				break;
+			case SRT_TO_UDP_TTL:
+				createStream.setSrtToUdpTtl(Integer.parseInt(value));
+				addAdvanceControlProperties(advancedControllableProperties,
+						createNumeric(stats, streamControllingGroup + StreamControllingMetric.SRT_TO_UDP_TTL.getName(), checkForNullData(createStream.getSrtToUdpTtl().toString())));
```

Comment:
can getSrtToUdpTtl() be null? Please check other toString() calls on dtos methods

Reply:
I fixed it, it can't be null any more